File: src/main/java/com/elchologamer/userlogin/util/database/sql/MySQL.java
package com.elchologamer.userlogin.util.database.sql;

import org.bukkit.configuration.ConfigurationSection;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.util.Properties;

public class MySQL extends SQLDatabase {

    public MySQL() {
        super("mysql", "com.mysql.jdbc.Driver");
    }

    @Override
    public void connect() throws SQLException, ClassNotFoundException {
        super.connect();

        // Remove username column if exists
        if (query("SHOW COLUMNS FROM " + getTable() + " LIKE 'username'").next()) {
            update("ALTER TABLE " + getTable() + " DROP COLUMN username");
        }

        // Update password max length
        int maxChars = Math.max(
                getPlugin().getConfig().getInt("password.maxCharacters", 128),
                1
        );
        update("ALTER TABLE " + getTable() + " MODIFY password VARCHAR(" + maxChars + ")");
    }

    @Override
    protected Connection getConnection() throws SQLException {
        ConfigurationSection section = getPlugin().getConfig()
                .getConfigurationSection("database.mysql");
        assert section != null;

        String host = section.getString("host", "localhost");
        int port = section.getInt("port", 3306);
        boolean ssl = section.getBoolean("ssl", section.getBoolean("useSSL"));

        String username = section.getString("username", "root");
        String password = section.getString("password", null);

        Properties props = new Properties();

        props.setProperty("user", username);
        if (password != null) props.setProperty("password", password);

        props.setProperty("useSSL", Boolean.toString(ssl));
        props.setProperty("autoReconnect", "true");

        String url = "jdbc:mysql://" + host + ":" + port + "/" + getDatabase();
        return DriverManager.getConnection(url, props);
    }
}

File: src/main/kotlin/com/elchologamer/userlogin/database/sql/MySQL.kt
package com.elchologamer.userlogin.database.sql

import com.elchologamer.userlogin.UserLogin.Companion.plugin
import org.bukkit.configuration.ConfigurationSection
import java.sql.Connection
import java.sql.DriverManager
import java.sql.SQLException
import java.util.*

class MySQL : SQLDatabase("mysql", "com.mysql.jdbc.Driver") {
    @Throws(SQLException::class, ClassNotFoundException::class)
    override fun connect() {
        super.connect()

        // Remove username column if exists
        if (query("SHOW COLUMNS FROM $table LIKE 'username'").next()) {
            update("ALTER TABLE $table DROP COLUMN username")
        }

        // Update password max length
        val maxChars = plugin.config.getInt("password.maxCharacters", 128).coerceAtLeast(1)
        update("ALTER TABLE $table MODIFY password VARCHAR($maxChars)")
    }

    override fun getConnection(): Connection {
        val section: ConfigurationSection = plugin.config.getConfigurationSection("database.mysql")
            ?: throw IllegalStateException("MySQL settings missing in config.yml")

        val host = section.getString("host", "localhost")
        val port = section.getInt("port", 3306)
        val ssl = section.getBoolean("ssl", section.getBoolean("useSSL"))
        val username = section.getString("username", "root")
        val password = section.getString("password", null)

        val props = Properties()

        props.setProperty("user", username)
        password?.let { props.setProperty("password", it) }
        props.setProperty("useSSL", ssl.toString())
        props.setProperty("autoReconnect", "true")

        return DriverManager.getConnection("jdbc:mysql://$host:$port/$database", props)
    }
}